version/version_test.go

```diff
@@ -0,0 +1,108 @@
+package version
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func replaceVersion(v, vp string) (string, string) {
+	origV := Version
+	origVP := VersionPrerelease
+
+	Version = v
+	VersionPrerelease = vp
+
+	return origV, origVP
+}
+
+func TestGetVersion(t *testing.T) {
+	// This test cannot be parallelized because it messes with some global
+	// variables that determine the version information.
+	origVersion, origVersionPrerelease := replaceVersion("1.2.3", "")
+
+	// Test the general case
+	vi := GetVersion()
+	assert.Equal(t, "1.2.3", vi.Version)
+	assert.Equal(t, "", vi.VersionPrerelease)
+	assert.Equal(t, "", vi.VersionMetadata)
+	assert.Equal(t, "", vi.Revision)
+	assert.Equal(t, "", vi.BuildDate)
+
+	// Test the git describe case
+	origGitDescribe := GitDescribe
+	GitDescribe = "git-describe"
+	vi = GetVersion()
+	assert.Equal(t, "git-describe", vi.Version)
+
+	GitDescribe = origGitDescribe
+
+	replaceVersion(origVersion, origVersionPrerelease)
```

Comment:
This pattern feels a little fragile to me. What are your thoughts on having `replaceVersion()` return a closure that resets version information to what it was, and people can run it in a `defer`, similar to how cluster cleanup works?  So something like:

```go
cleanup := replaceVersion("1.2.3", "")
defer cleanup()
```